src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -245,51 +215,48 @@ export default class NativeAdapter implements IMediaSourceAdapter {
    * Select a video track
    * @function selectVideoTrack
    * @param {VideoTrack} videoTrack - the track to select
-   * @returns {boolean} - success
+   * @returns {void}
    * @public
    */
-  selectVideoTrack(videoTrack: VideoTrack): boolean {
+  selectVideoTrack(videoTrack: VideoTrack): void {
     let videoTracks = this._videoElement.videoTracks;
     if ((videoTrack instanceof VideoTrack) && videoTracks && videoTracks[videoTrack.index]) {
       this._disableVideoTracks();
       videoTracks[videoTrack.index].selected = true;
-      return true;
+      this._trigger(BaseMediaSourceAdapter.CustomEvents.VIDEO_TRACK_CHANGED, {selectedVideoTrack: videoTrack});
```

Comment:
what should be in the base?
the selection itself is an adapter unique logic.
the event trigger is in the base.
not sure I understand you here.